src/author.ts

```diff
@@ -0,0 +1,111 @@
+import { upperFirst } from "scule";
+import { ResolvedChangelogConfig } from "./config";
+import { GitCommit } from "./git";
+import { getGithubLoginByCommit } from "./github";
+
+export interface GithubAuthor {
+  commits: string[];
+  github?: string;
+  email: Set<string>;
+  name: string;
+}
+
+export async function resolveAuthorInfo(
+  config: ResolvedChangelogConfig,
+  info: GithubAuthor
+) {
+  if (info.github) {
+    return info;
+  }
+
+  try {
+    for (const email of info.email) {
+      const { user } = await fetch(`https://ungh.cc/users/find/${email}`)
+        .then((r) => r.json())
+        .catch(() => ({ user: null }));
+      if (user) {
+        info.github = user.username;
+        break;
+      }
+    }
+  } catch {}
+
+  if (info.github) {
+    return info;
+  }
+
+  // token not provided, skip github resolving
+  if (!config.tokens.github) {
+    return info;
+  }
+
+  if (info.commits.length > 0) {
+    try {
+      info.github = await getGithubLoginByCommit(config, info.commits[0]);
```

Comment:
I think here we can iterate over `commits` until finding first author (and then `break` loop)